index.js

```diff
@@ -1,11 +1,134 @@
-const express = require('express');
+//ASSIGNMENT: Use Node.js and Express to build an API that performs CRUD operations on users.
 
+//we use require() to import the express module and make it available to our application. This is similar to the import keyword we have used before. 
+const express = require('express'); 
 
 //The return of calling express() is an instance of an Express application that we can use to configure our server and, eventually, start “listening” and responding to requests. 
-const server = express(); 
+const server = express();
 
-server.get('/', (req, res) =>{
+// We are using the .get() method to set up a route handler function that will be executed on every GET request to the URL specified as the first parameter, in this case the root of the site (represented by a /). 
+
+// The first two arguments passed by express to a route handler function are: an object that represents the request and another object that represents the response. 
+// Express augments those objects with a set of useful properties and methods. Our example uses the .send() method of the response object to specify the data sent to the client as the response body. 
+
+server.get('/', (req, res) => {
     res.send('Hello World');
 });
 
-server.listen(8000, () => console.log('API running on port 8000'));
\ No newline at end of file
+let db = require('./data/db')
+
+
+
+// The .find() method returns a promise, so make sure to send 
+// the response after that promise has resolved and, in case 
+// of failure, return a status code of 500 and an error message 
+// back to the client.
+//
+// function find() {
+//     return db('users');
+//   }
+
+// server.get('/api/users', (req,res) => {
+//     let users = db.find()
+//     res.send(users);
+// })
+
+const promise = db.find()
+
+
+
+server.get('/api/users', (req,res) => {
+    db.find()
```

Comment:
So, here you will only need to call the method of the database once THEN you will simple send back the users. After that, you would just need to set-up a catch for error handling.